src/component/mxgraph/shape/IconPainter.ts

```diff
@@ -84,179 +114,175 @@ export default class IconPainter {
 
   // this implementation is adapted from the draw.io BPMN 'message' symbol
   // https://github.com/jgraph/drawio/blob/0e19be6b42755790a749af30450c78c0d83be765/src/main/webapp/shapes/bpmn/mxBpmnShape2.js#L465
-  public static paintEnvelopIcon({ c, shape: { x, y, w, h }, icon }: PaintParameter): void {
-    this.updateCanvasStyle(c, icon);
+  public static paintEnvelopIcon({ canvas, ratioFromParent, shape, icon }: PaintParameter): void {
+    this.updateCanvasStyle(canvas, icon);
 
-    // Change the coordinate referential
-    c.translate(x + w * 0.24, y + h * 0.34);
-    w = w * 0.52;
-    h = h * 0.32;
+    const initialIconSize = { width: 485.41, height: 321.76 };
+    const iconSize = this.calculateIconSize(initialIconSize, icon, shape, ratioFromParent);
+    this.translateIconToShapeCenter(canvas, shape, iconSize.width, iconSize.height);
 
     // Paint the envelope outline with dark color
-    c.rect(0, 0, w, h);
-    c.fillAndStroke();
+    canvas.rect(0, 0, iconSize.width, iconSize.height);
```

Comment:
:question: why not introduce w (iconSize.width) and h (iconSize.height) constants to keep the code in place?

Reply:
I can do this, but at a moment, I did'nt know what I manipulated the width of the shape or the width of the icon (after its resizing)